source/common/network/io_uring_socket_handle_impl.cc

```diff
@@ -0,0 +1,415 @@
+  ENVOY_LOG(trace, "write {}, fd = {}, type = {}", buffer.length(), fd_, ioUringSocketTypeStr());
+
+  auto write_result = checkWriteResult();
+  if (write_result.has_value()) {
+    return std::move(*write_result);
+  }
+
+  uint64_t buffer_size = buffer.length();
+  io_uring_socket_->write(buffer);
+  return {buffer_size, IoSocketError::none()};
+}
+
+Api::IoCallUint64Result IoUringSocketHandleImpl::sendmsg(const Buffer::RawSlice*, uint64_t, int,
+                                                         const Address::Ip*,
+                                                         const Address::Instance&) {
+  ENVOY_LOG(trace, "sendmsg, fd = {}, type = {}", fd_, ioUringSocketTypeStr());
+  return Network::IoSocketError::ioResultSocketInvalidAddress();
+}
+
+Api::IoCallUint64Result IoUringSocketHandleImpl::recvmsg(Buffer::RawSlice*, const uint64_t,
+                                                         uint32_t, RecvMsgOutput&) {
+  ENVOY_LOG(trace, "recvmsg, fd = {}, type = {}", fd_, ioUringSocketTypeStr());
+  return Network::IoSocketError::ioResultSocketInvalidAddress();
+}
+
+Api::IoCallUint64Result IoUringSocketHandleImpl::recvmmsg(RawSliceArrays&, uint32_t,
+                                                          RecvMsgOutput&) {
+  ENVOY_LOG(trace, "recvmmsg, fd = {}, type = {}", fd_, ioUringSocketTypeStr());
+  return Network::IoSocketError::ioResultSocketInvalidAddress();
+}
+
+Api::IoCallUint64Result IoUringSocketHandleImpl::recv(void* buffer, size_t length, int flags) {
+  ASSERT(io_uring_socket_.has_value());
+  ENVOY_LOG(trace, "recv, fd = {}, type = {}", fd_, ioUringSocketTypeStr());
+
+  // The only used flag in Envoy is MSG_PEEK for listener filters, including TLS inspectors.
+  ASSERT(flags == 0 || flags == MSG_PEEK);
+  Buffer::RawSlice slice;
+  slice.mem_ = buffer;
+  slice.len_ = length;
+  if (flags == 0) {
+    return readv(length, &slice, 1);
+  }
+
+  return copyOut(length, &slice, 1);
+}
+
+Api::SysCallIntResult IoUringSocketHandleImpl::bind(Address::InstanceConstSharedPtr address) {
+  ENVOY_LOG(trace, "bind {}, fd = {}, io_uring_socket_type = {}", address->asString(), fd_,
+            ioUringSocketTypeStr());
+  return Api::OsSysCallsSingleton::get().bind(fd_, address->sockAddr(), address->sockAddrLen());
+}
+
+Api::SysCallIntResult IoUringSocketHandleImpl::listen(int backlog) {
+  ENVOY_LOG(trace, "listen, fd = {}, type = {}", fd_, ioUringSocketTypeStr());
+
+  ASSERT(io_uring_socket_type_ == IoUringSocketType::Unknown);
+
+  io_uring_socket_type_ = IoUringSocketType::Accept;
+  setBlocking(false);
+  return Api::OsSysCallsSingleton::get().listen(fd_, backlog);
+}
+
+IoHandlePtr IoUringSocketHandleImpl::accept(struct sockaddr* addr, socklen_t* addrlen) {
+  ENVOY_LOG(trace, "accept, fd = {}, type = {}", fd_, ioUringSocketTypeStr());
+
+  ASSERT(io_uring_socket_type_ == IoUringSocketType::Accept);
+
+  auto result = Api::OsSysCallsSingleton::get().accept(fd_, addr, addrlen);
```

Comment:
Full type please.

Reply:
Well, if you want the full type, then I can just give you the full type. But I think the result from `accept` is also `auto` even in the core io_socket_handle_impl.cc.